### PR TITLE
rfcs: graph api: add python binding

### PR DESCRIPTION
**Pull request by TaoLv**
This is to propose adding python binding for onDNN graph API.

The rendered document can be found at [link](https://github.com/TaoLv/mkl-dnn/blob/lvtao/rfc/python-api/rfcs/20230720-graph-python-api/README.md).

### CODE REVIEW

**Comment by wenzhe-nrv** [comment]
LGTM overall. 

Few notes:
* I wonder how to integrate with the current installation setup, which requires the user to `pip install`. 
* For the current code, I would recommend to consider adding default args for APIs as the c++ API. E.g. logical tensor, I found that building graph without concrete shape/stride, and later updating those in runtime with real tensor helpful. 
* For tensor pointers, this works for PyTorch
  ```
  .def(pybind11::init(
            [](logical_tensor& lt, engine& eng, uint64_t data_ptr) {
              return tensor(lt, eng, reinterpret_cast<void*>(data_ptr));
            }))
  ```

**Comment by sanchitintel** [comment]
We will put the bindings in [`ideep`](https://github.com/intel/ideep/)

File: rfcs/20230720-graph-python-api/README.md [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+    into its 3rd party in order to use the python binding.
+
+- Option 3: put the source code of graph API python binding as part of PyTorch.
+  - Pros: currently the main request of graph API python binding is from PyTorch,
+    PyTorch team already has the implementation in the PR. This option also has
+    less impact to oneDNN, and there's no need to start a new repository.
+  - Cons: more code to upstream to PyTorch, which leads to concerns from the community.
+
+As the request from PyTorch is to support option 1, below discusses more details
+about option 1.
```
Comment (sanchitintel) [inline rfcs/20230720-graph-python-api/README.md:135]: We will put the bindings in [`ideep`](https://github.com/intel/ideep/). Thanks!

**Comment by ElaineBao** [comment]
> Hi @ElaineBao, can you please clarify how oneDNN Graph python binding code would be updated based on API changes?
> 
> If you wouldn't have a dedicated GitHub repository for them, should we port your code to ideep, and then modify bindings as & when required? Thanks

Hi, @sanchitintel , as your comment mentioned, if oneDNN Graph python binding will be put in ideep, then whenever ideep upgrades oneDNN & oneDNN Graph has API changes, the python binding in ideep need to be updated.
Thanks.

**Comment by sanchitintel** [comment]
Thanks for the info, @ElaineBao! 
Would you like to port it to ideep, since your team authored the code? Prior git history would be lost while porting over.

**Review by igorsafo — APPROVED** [review]

**Comment by igorsafo** [comment]
@ElaineBao & @TaoLv Thank you for gathering all information and discussions into the RFC! This RFC will be very helpful in case we decide to change the direction in future.

**Comment by ElaineBao** [comment]
> Thanks for the info, @ElaineBao! Would you like to port it to ideep, since your team authored the code? Prior git history would be lost while porting over.

Sorry, I forgot to reply here.. As discussed offline, we are glad to help port the python binding to ideep in couple of weeks!

**Review by sanchitintel — APPROVED** [review]
The RFC LGTM. Thanks for providing implementation code as well!

**Comment by TaoLv** [comment]
Thank you @ElaineBao @sanchitintel @igorsafo. I'm merging this RFC for archiving.

**Pull request merged**